pkg/gardenlet/operation/types.go

```diff
@@ -28,12 +28,13 @@ import (
 // Builder is an object that builds Operation objects.
 type Builder struct {
 	configFunc                func() (*gardenletconfigv1alpha1.GardenletConfiguration, error)
-	gardenFunc                func(context.Context, map[string]*corev1.Secret, *gardenerutils.Domain) (*garden.Garden, error)
+	gardenFunc                func(context.Context, *gardenerutils.Domain, []*gardenerutils.Domain) (*garden.Garden, error)
```

Comment:
It would be nice to document that the second parameter is the internal domain, while the third one is the default domains. Maybe just 

```suggestion
	gardenFunc                func(ctx context.Context, internalDomain *gardenerutils.Domain, defaultDomains []*gardenerutils.Domain) (*garden.Garden, error)
```

would be enough (this requires the context parameter to be also named).